apps/checkout/src/lib/utils/address.ts

```diff
@@ -1,24 +1,59 @@
-import { AddressField } from "@/lib/globalTypes";
+import uniq from "lodash/uniq";
+import compact from "lodash/compact";
+import { AddressField, ApiAddressField } from "../globalTypes";
 
 const addressFieldsOrder: AddressField[] = [
-  "name",
+  "firstName",
+  "lastName",
   "companyName",
+  "phone",
   "streetAddress1",
   "streetAddress2",
   "city",
   "postalCode",
+  "cityArea",
+  "countryArea",
 ];
 
+// api doesn't approve of "name" so we replace it with "firstName"
+// and "lastName"
+export const getFilteredAddressFields = (
+  addressFields: ApiAddressField[]
+): AddressField[] => {
+  const filteredAddressFields = addressFields.filter(
+    (addressField: ApiAddressField) => addressField !== "name"
+  ) as AddressField[];
+
+  return uniq([...filteredAddressFields, "firstName", "lastName"]);
+};
+
 // api doesn't order the fields but we want to
-export const getSortedAddressFields = (addressFields: AddressField[] = []) =>
-  addressFieldsOrder.reduce((result, orderedAddressField) => {
-    if (!addressFields.includes(orderedAddressField)) {
+export const getSortedAddressFields = (addressFields: AddressField[] = []) => {
+  const filteredAddressFields = getFilteredAddressFields(addressFields);
+
+  return addressFieldsOrder.reduce((result, orderedAddressField) => {
+    if (!filteredAddressFields.includes(orderedAddressField)) {
       return result;
     }
 
     return [...result, orderedAddressField];
   }, [] as AddressField[]);
+};
 
 export const getSortedAddressFieldsFromAddress = (
   address: Partial<Record<AddressField, any>>
 ) => getSortedAddressFields(Object.keys(address) as AddressField[]);
+
+export const ensureArray = <TData extends any[] | null[] | null>(
```

Comment:
What do we need this function for? Do the lists have falsy values in them that we want to filter out?

Reply:
We have plenty of places in the api where it's either the data we want or null or undefined or null[] so... yeah, we're using currently in `src/sections/UserAddresses/AddressForm.tsx`

Reply:
`null[]` should no longer be true after recent core changes - https://github.com/saleor/saleor/pull/9391
seems like `|| []` would suffice